scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/system_commands.py

```diff
@@ -13,64 +13,29 @@
 #    See the License for the specific language governing permissions and
 #    limitations under the License.
 
+import filecmp
+import os
+import subprocess
 import sys
-import xmlrpc.client
 
 from ..pseudo_cluster import PseudoCluster
+from .accessory_server_bridge import AccessoryServerBridge
 
-DEFAULT_KEY = 'default'
-IP = '127.0.0.1'
-PORT = 9000
+DEFAULT_CHIP_ROOT = os.path.abspath(
```

Comment:
I told you that it should be a package ;) 
More seriously, that is annoying. I think this is OK for now as we need to move forward but we need to find a proper solution at the end of the day.

Reply:
Even if this was it's own package of yamlclusters you would still need to solve this are you are grabbing a file that is outside of the packages using a relative path that is outside of `<chip_root>/scripts`

If we are time bound we could let this in as is, but it will bite us later on when TH team is trying to port this using package. But, I do think that this can be fixed with 1 hour of work to properly allow for code reuse and python code package distribution either:
1. Package `/src/app/ota_image_tool.py` to be it's own package which yamltests will depend on.
2. Take out the required code that both sides need and place that in its own package inside scripts, and have both that script and yamltests depend on this new package.

Reply:
Wait I just realized you are using relative path based on where this happens to get installed in the virtual enviroment. This is not even relative to where this file sits in the repo here. I think we need to fix this before merging

Reply:
if we run under `bootstrap/activate` then `$PW_PROJECT_ROOT` is set. Or otherwise lets make some env variable or command line argument override available.

Reply:
I would suggest some `_GetChipRoot` that checks various env variables (maybe a dedicated one, followed by a PW_PROJECT_ROOT fallback and if that fails, relative path). I would also check for some known file existence and print out a friendly exception if the check fails.

Allow the user to bypass things.

Reply:
Sorry for the lag on this. I will come up with something in the next few days.

Reply:
Ok. I moved the related OTA specific function code under `scripts/tests/chiptest` for 2 reasons.
The first one is to avoid duplicating the specific `CompareFiles.py` and `CreateOTAImage.py` code.
The second is because in my case the test harness (in this case `scripts/tests/chiptest`) can be remote, so it needs to run on the remote.